zetaclient/btc_signer.go

```diff
@@ -67,8 +67,13 @@ func NewBTCSigner(cfg config.BTCConfig, tssSigner TSSSigner, logger zerolog.Logg
 
 // SignWithdrawTx receives utxos sorted by value, amount in BTC, feeRate in BTC per Kb
 func (signer *BTCSigner) SignWithdrawTx(to *btcutil.AddressWitnessPubKeyHash, amount float64, gasPrice *big.Int, sizeLimit uint64,
-	btcClient *BitcoinChainClient, height uint64, nonce uint64, chain *common.Chain) (*wire.MsgTx, error) {
+	btcClient *BitcoinChainClient, height uint64, nonce uint64, chain *common.Chain, cointype common.CoinType) (*wire.MsgTx, error) {
+
 	estimateFee := float64(gasPrice.Uint64()) * outTxBytesMax / 1e8
+	if cointype == common.CoinType_Cmd {
```

Comment:
I think for real Tss migration (TSS had 3 hundreds of UTXOs in Athens3), if we pays only `outTxBytesMin` of fee, it may take many blocks (10min per block) for the outTx (many many inputs) to be accepted in a block.
The thing for TSS migration here is to make sure the old tss address has enough funds `>= 80_000 sats (4000 * gasPrice (e.g., 20))` to move its own funds.
Also, I believe the hard coded `estimateFee = 4000` can be improved in future too, but that only helps  in cases when TSS has only $10 (for example) of BTC in reserve and we want to move $5 to new TSS and paying another $5 as gas fee.
To conclude, we can just remove the `estimateFee` adjustment.

Reply:
Not sure I understand completely, you mean to remove the estimate fee component completely?

Reply:
I think comment out line 74 and leave the calculation `estimateFee := float64(gasPrice.Uint64()) * outTxBytesMax / 1e8` as is would be enough.